platform-metrics/src/main/java/org/hypertrace/core/serviceframework/metrics/PlatformMetricsRegistry.java

```diff
@@ -375,9 +377,18 @@ public static DistributionSummary registerDistributionSummary(String name,
   }
 
   /**
-   * Registers metrics for the given executor service with the service's metric registry and
-   * reports them periodically to the configured reporters. Apart from the given tags, the
-   * reporting service's default tags also will be reported with the metrics.
+   * Registers metrics for GuavaCaches using micrometer's GuavaCacheMetrics under the given
+   * cacheName for the given guavaCache
+   */
+  public static <K, V> Cache<K, V> registerCache(String cacheName, Cache<K, V> guavaCache) {
+    Cache<K, V> monitor = GuavaCacheMetrics.monitor(METER_REGISTRY, guavaCache, cacheName);
```

Comment:
Why do we have to return.
Are consumers expected to use the returned cache from here? I doubt so.

Reply:
No not really, both the cache's are identical. Let me fix this.